superset-frontend/src/dashboard/util/charts/getFormDataWithExtraFilters.ts

```diff
@@ -68,13 +69,15 @@ export default function getFormDataWithExtraFilters({
 
   // if dashboard metadata + filters have not changed, use cache if possible
   if (
-    (cachedFiltersByChart[sliceId] || {}) === filters &&
-    (colorScheme == null ||
-      cachedFormdataByChart[sliceId].color_scheme === colorScheme) &&
-    cachedFormdataByChart[sliceId].color_namespace === colorNamespace &&
-    isEqual(cachedFormdataByChart[sliceId].label_colors, labelColors) &&
+    cachedFiltersByChart[sliceId] === filters &&
+    (colorScheme === null ||
+      cachedFormdataByChart[sliceId]?.color_scheme === colorScheme) &&
+    cachedFormdataByChart[sliceId]?.color_namespace === colorNamespace &&
+    isEqual(cachedFormdataByChart[sliceId]?.label_colors, labelColors) &&
     !!cachedFormdataByChart[sliceId] &&
-    dataMask === undefined
+    areObjectsEqual(cachedFormdataByChart[sliceId]?.dataMask, dataMask, {
+      ignoreUndefined: true,
+    })
```

Comment:
While we're modifying code here and for the sake of DRY, would it make sense to break out
```js
const cachedFormdata = cachedFormdataByChart[sliceId];
```